File: core/src/main/java/io/github/fourlastor/game/coordinates/Packer.java
package io.github.fourlastor.game.coordinates;

import com.badlogic.gdx.math.GridPoint2;

public class Packer {

    public static int pack(GridPoint2 position) {
        return position.x << 16 | position.y;
    }

    public static GridPoint2 unpack(int value, GridPoint2 target) {
        return target.set(value >> 16, (short) value);
    }
}

File: core/src/main/java/io/github/fourlastor/game/coordinates/Packer.kt
package io.github.fourlastor.game.coordinates

import com.badlogic.gdx.math.GridPoint2

object Packer {
  fun pack(position: GridPoint2): Int {
    return position.x shl 16 or position.y
  }

  fun unpack(value: Int, target: GridPoint2): GridPoint2 {
    return target.set(value shr 16, value.toShort().toInt())
  }
}
